scripts/hf_example.py

```diff
@@ -0,0 +1,109 @@
+import argparse
+import transformers
+from fms.models import llama
+import os
+import time
+import torch
+from torch import nn
+from fms.models.hf.llama import modeling_llama_hf
+from transformers import pipeline
+from fms.models.hf.utils import register_fms_models
+
+# This script demonstrates how to use FMS model implementations with HF formatted
+# weights.
+#
+# Requires first installing transformers, sentencepiece, protobuf, and torch >= 2.1.0
+
+parser = argparse.ArgumentParser(
+    description="Example script to load HF weights into an FMS model, and use them in the HF ecosystem"
+)
+
+parser.add_argument(
+    "--model_path",
+    type=str,
+    required=True,
+    help="The path to a directory containing hugging-face formatted LLaMA weights and tokenizer",
+)
+parser.add_argument(
+    "--prompt",
+    type=str,
+    default="""q: how are you? a: I am good. How about you? q: What is the weather like today? a:""",
+    help="An input prompt to seed output from the LM",
+)
+
+args = parser.parse_args()
+
+local_rank = int(os.environ.get("LOCAL_RANK", 0))
+device = torch.device("cuda", local_rank)
+
+tokenizer = transformers.LlamaTokenizer.from_pretrained(args.model_path)
+prompt = args.prompt
+prompt_tokens = tokenizer.tokenize(prompt)
+input_ids = tokenizer.convert_tokens_to_ids(prompt_tokens)
+input_ids = torch.tensor(input_ids, dtype=torch.long, device=device).unsqueeze(0)
+
+# Create an instance of the huggingface model using huggingface weights
+model = transformers.LlamaForCausalLM.from_pretrained(
+    args.model_path, use_safetensors=True
+)
+model.to(device)
+model = model.to(torch.half)
+
+
+# make sure we always call generate the same way when comparing implementations.
+def generate(model: nn.Module, new_tokens=25):
+    max_length = input_ids.shape[1] + new_tokens
+    return model.generate(
+        input_ids=input_ids, use_cache=True, max_length=max_length, do_sample=False
```

Comment:
might want to consider using max_new_tokens=new_tokens instead of adding the input_ids shape to new tokens and passing as max_length

Reply:
I can't find anything clear and definitive about `max_new_tokens` in the source code nor docs:

https://github.com/huggingface/transformers/blob/c4d4e8bdbd25d9463d41de6398940329c89b7fb6/src/transformers/generation_utils.py#L101

code that I did find for handling max_new_tokens appeared not to match what we'd want it to do in this case, but I can't remember where I found that code.

Reply:
The docs can be found here https://huggingface.co/docs/transformers/main_classes/text_generation#transformers.GenerationConfig.max_new_tokens

Reply:
updated.

it's odd that the doc isn't on the actual function, and GenerationConfig isn't super clear that it's actually the doc for that generate function (as opposed to pipelines and other ways to generate). it seemed safer to just use the documented flags but ... this is fine too.